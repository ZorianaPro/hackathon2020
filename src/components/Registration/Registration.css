.Registration {
    background-color: #5f4b8b;
}

.Registration .Registration-Container {
    padding: 4em 0;
    margin: 0 auto;
    height: 100%;
    position: relative;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    max-width: var(--maxWidth);
}

.Registration .Registration-Title {
    font-size: var(--fontSize48);
    font-weight: bold;
    font-style: italic;
    line-height: var(--lineHeight);
    letter-spacing: normal;
    text-align: center;
    color: white;
}

.Registration .Registration-Content {
    width: 100%;
    display: block;
    text-align: center;
    margin-top: 4em;
}

.Registration .Button {
    display: inline-block;
    margin-left: -140px;
}

@media screen and (max-width: 894px) {
    .Registration .Button {
        margin: 0 0 1em -89px;
    }
}