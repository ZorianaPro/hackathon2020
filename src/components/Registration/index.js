import React, { useCallback } from 'react';
import './Registration.css';
import Button from '../Button';

const Registration = ({
  onOpenAddIdeaOverlay,
  onOpenJoinIdeaOverlay
}) => {
  const _onOpenJoinIdeaOverlay = useCallback((event) => {
    if (typeof onOpenJoinIdeaOverlay === 'function') {
      onOpenJoinIdeaOverlay(event);
    }
  }, [
    onOpenJoinIdeaOverlay
  ]);

  const _onOpenAddIdeaOverlay = useCallback((event) => {
    if (typeof onOpenAddIdeaOverlay === 'function') {
      onOpenAddIdeaOverlay(event);
    }
  }, [
    onOpenAddIdeaOverlay
  ]);

  return (
    <section id="registration" className="Registration">
      <div className="Registration-Container">
        <p className="Registration-Title">
          Registration
        </p>
        <div className="Registration-Content">
          <Button label="Join an existing idea"
            type="isActionFilled"
            onClick={
              _onOpenJoinIdeaOverlay
            }
          />
          <Button label="Create your own idea"
            type="isActionFilled"
            onClick={
              _onOpenAddIdeaOverlay
            }
          />
        </div>
      </div>
    </section>
  );
};

export default Registration;