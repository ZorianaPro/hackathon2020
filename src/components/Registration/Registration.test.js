import React from 'react';
import { shallow, mount } from 'enzyme';
import Registration from './';

describe('Registration', () => {
  it('renders without crashing', () => {
    expect(shallow.bind(shallow, <Registration/>)).not.toThrow();
  });

  it('renders as expected', () => {
    expect(mount(<Registration/>)).toMatchSnapshot();
  });
});