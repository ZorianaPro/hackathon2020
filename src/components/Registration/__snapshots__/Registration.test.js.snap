// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration renders as expected 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Registration>
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <section
          className="Registration"
          id="registration"
        >
          <div
            className="Registration-Container"
          >
            <p
              className="Registration-Title"
            >
              Registration
            </p>
            <div
              className="Registration-Content"
            >
              <Button
                label="Join an existing idea"
                onClick={[Function]}
                type="isActionFilled"
              >
                <div
                  className="Button Button--isActionFilled"
                >
                  <p
                    className="Button-Label"
                    onClick={[Function]}
                  >
                    Join an existing idea
                  </p>
                  <div
                    className="Button-Inside"
                    onClick={[Function]}
                  />
                </div>
              </Button>
              <Button
                label="Create your own idea"
                onClick={[Function]}
                type="isActionFilled"
              >
                <div
                  className="Button Button--isActionFilled"
                >
                  <p
                    className="Button-Label"
                    onClick={[Function]}
                  >
                    Create your own idea
                  </p>
                  <div
                    className="Button-Inside"
                    onClick={[Function]}
                  />
                </div>
              </Button>
            </div>
          </div>
        </section>
      </Router>
    </BrowserRouter>
  </Registration>
</Provider>
`;
